src/main/java/uk/gov/pay/products/persistence/entity/ProductMetadataEntity.java

```diff
@@ -13,9 +13,9 @@
 @Table(name = "products_metadata")
 public class ProductMetadataEntity extends AbstractEntity {
 
-    @ManyToOne(cascade = CascadeType.MERGE)
-    @JoinColumn(name = "product_id", updatable = false)
-    private ProductEntity productId;
+    @ManyToOne(fetch = FetchType.EAGER)
```

Comment:
any reason this is EAGER ? We seem to always get productEntity first to start processing (querying) metadata. Not sure if JPA issues additional queries to get productEntity for each metadata row

Reply:
no reason, I can remove it if you think that's a better idea